perl/modules/Zulip/lib/BarnOwl/Module/Zulip.pm

```diff
@@ -0,0 +1,522 @@
+            $tls_ctx = new AnyEvent::TLS(verify => $cfg{'ssl_verify'}, 
+                                         sslv3 => 0, verify_peername => "http",
+                                         ca_file => $cfg{'ssl_ca_file'},
+                                         cert_file => $cfg{'ssl_cert_file'},
+                                         key_file => $cfg{'ssl_key_file'});
+        } else {
+            $tls_ctx = new AnyEvent::TLS(verify => $cfg{'ssl_verify'}, 
+                                         sslv3 => 0, verify_peername => "http",
+                                         ca_file => $cfg{'ssl_ca_file'});
+        }           
+    } else {
+        # we still want it for a unique id
+        $tls_ctx = int(rand(1024));
+    }
+}
+
+
+sub authorization {
+    return "Basic " . encode_base64($cfg{'user'} . ':' . $cfg{'apikey'}, "")
+}
+
+sub read_config {
+    my $fh = shift;
+
+    my $raw_cfg = do {local $/; <$fh>};
+    close($fh);
+    eval {
+        $raw_cfg = from_json($raw_cfg);
+    };
+    if($@) {
+        fail("Unable to parse config file: $@");
+    }
+
+    if(! exists $raw_cfg->{user}) {
+        fail("Account has no username set.");
+    }
+    my $user = $raw_cfg->{user};
+    if(! exists $raw_cfg->{apikey}) {
+        fail("Account has no api key set.");
+    }
+    my $apikey = $raw_cfg->{apikey};
+    if(! exists $raw_cfg->{api_url}) {
+        fail("Account has no API url set.");
+    }
+    my $api_url = $raw_cfg->{api_url};
+
+    if(! exists $raw_cfg->{default_realm}) {
+        fail("Account has no default realm set.");
+    }
+    my $default_realm = $raw_cfg->{default_realm};
+
+    if( exists $raw_cfg->{ssl}) {
+        # mandatory parameters
+        if (! exists $raw_cfg->{ssl}->{ca_file}) {
+            fail("SSL parameters specified, but no CA file set");
+        }
+        $cfg{'ssl_ca_file'} = $raw_cfg->{ssl}->{ca_file};
+        $cfg{'ssl_verify'} = 1;
+        # optional parameters
+        if ( (exists $raw_cfg->{ssl}->{cert_file}) && exists $raw_cfg->{ssl}->{key_file}) {
+            $cfg{'ssl_cert_file'} = $raw_cfg->{ssl}->{cert_file};
+            $cfg{'ssl_key_file'} = $raw_cfg->{ssl}->{key_file};
+        }  else {
+            warn "SSL parameters specified, but no client credentials set.";
+        }
+    } else {
+	$cfg{'ssl_verify'} = 0;
+	my $msg = "SSL parameters not specified. WILL NOT VERIFY SERVER CERTIFICATE. See README for details.";
+	BarnOwl::admin_message('Zulip Warning', "Zulip: $msg");
+	warn $msg;
+    }
+    
+    $cfg{'user'} = $user;
+    $cfg{'apikey'} = $apikey;
+    $cfg{'api_url'} = $api_url;
+    $cfg{'realm'} = $default_realm;
+}
+
+sub register {
+    my $retry_count = 0;
+    my $callback;
+
+    if(!exists $cfg{'api_url'}) {
+        die("Zulip not configured, cannot poll for events");
+    }
+    $callback = sub {
+        BarnOwl::debug("In register callback");
+        my ($body, $headers) = @_;
+        if($headers->{Status} > 399) {
```

Comment:
Are you expecting anything other than 200 here (and the other instances of `> 399` and `< 400`)? `!= 200` seems safer.